Gems/GradientSignal/Code/Source/Editor/EditorGradientBakerComponent.cpp

```diff
@@ -507,6 +520,23 @@ namespace GradientSignal
 
             AZ::TickBus::Handler::BusDisconnect();
 
+            // After a successful bake, if the Entity that contains this gradient baker component also
+            // has an image gradient component, then update the image gradient's image asset with the
+            // output path that we baked to
+            if (AzToolsFramework::EntityHasComponentOfType(GetEntityId(), GradientSignal::EditorImageGradientComponentTypeId))
+            {
```

Comment:
There is currently only one image gradient component. It's not likely that there will be another implementation. However, in the event that another image gradient component implementation is developed, it might be worth checking if the entity is connected to GradientSignal::ImageGradientRequestBus instead of checking for the specific component type.

Reply:
Actually, I'll throw one more suggestion out there, since it's something that might come in handy in a few other places. It would really be nice to have a helper method to detect if an entity has a component that implements a specific service. This would ideally be in AzCore / AzFramework, not AzToolsFramework, so that we could use it at runtime. I've been thinking this might be a way to do some smarter component initialization that avoids extra refreshes. Some examples:
* Terrain Physics Collider could detect when it's on an entity with a terrain spawner, and defers initialization because it knows it will get a refresh from the spawner.
* Mesh component could detect when it's on an entity with a material component, and defer initialization until it gets the material refresh.

I'm not _sure_ these are a good idea in practice, but if a helper method like that helps in this case too so that you can detect a component by service instead of type ID, then I figured I'd throw it out there. :)

Reply:
Though thinking about it more, looking for the request bus handler is probably a whole lot easier and more straightforward in this case. Never mind. :)

Reply:
Agreed! Discussed Mike's idea offline (I actually was initially looking for a helper that would do just that, to be able to find a component on an entity that provided a service). I agree with y'all that using the `GradientSignal::ImageGradientRequestBus::HasHandlers` in this case would be the most straightforward. Will keep the helper for querying based on a service in the back of my mind.